test_tools/src/test_tools/learn_objects.py

```diff
@@ -0,0 +1,73 @@
+#! /usr/bin/env python
+
+import sys
+
+import rospy
+import PyKDL as kdl
+
+
+from robot_smach_states.util.designators import EntityByIdDesignator
+from robot_skills.util.kdl_conversions import VectorStamped
+from robot_skills.util.robot_constructor import robot_constructor
+
+
+def look_at_entity(robot, location_des):
+    # look on top of the surface
+    entity = location_des.resolve()
+    area = "on_top_of"
+    search_volume = entity.volumes[area]
+    x_obj = 0.5 * (search_volume.min_corner.x() + search_volume.max_corner.x())
+    y_obj = 0.5 * (search_volume.min_corner.y() + search_volume.max_corner.y())
+    z_obj = search_volume.min_corner.z()
+    lookat_pos_map = entity.pose.frame * kdl.Vector(x_obj, y_obj, z_obj)
+    x = lookat_pos_map.x()
+    y = lookat_pos_map.y()
+    z = lookat_pos_map.z()
+
+    robot.head.look_at_point(VectorStamped(x, y, z, "/map"), timeout=0)
+    rospy.loginfo("Looking at position ({}, {}, {})".format(x, y, z))
+    robot.head.wait_for_motion_done()
+    rospy.loginfo("done looking a the {}".format(entity.id))
+
+
+if __name__ == '__main__':
+    rospy.init_node('learn_objects')
+
+    if len(sys.argv) < 2:
+        print("Please specify a robot name")
+        sys.exit()
+
+    robot_name = sys.argv[1]
+    if robot_name != "hero":
+        rospy.logwarn("Currently learn_objects is designed to be used with hero. It will not work for other robots")
+        sys.exit()
+
+    location = "hero_case"
+
+    robot = robot_constructor(robot_name)
+
+    location_des = EntityByIdDesignator(robot, location)
+
+    arm = robot.leftArm()
+    arm.send_joint_goal('arm_out_of_way')
+    arm.wait_for_motion_done()
+
+    robot.torso.high()
+    robot.torso.wait_for_motion_done()
+
+    look_at_entity(robot, location_des)
+
+    entity = location_des.resolve()
+    while not rospy.is_shutdown():
+        count = 0
+        while not rospy.is_shutdown() and count < 10:
+            robot.speech.speak("Say Cheese")
+            res = robot.ed.update_kinect("{} {}".format("on_top_of", entity.id))
+            segmented = res.new_ids + res.updated_ids
+            object_classifications = robot.ed.classify(ids=segmented)
+            rospy.sleep(2)
+            count += 1
+        robot.speech.speak("Pause")
+        rospy.sleep(5)
```

Comment:
During the annotating I also saw that there were quite a few pictures with a hand/movement in them so it might be smart to make this sleep a little longer

Reply:
The difficult part is that it is hard to tell when the robot is taking pictures. This is due to this being a first version. Using the turning disk should allow us to make pictures faster and without hands in the way.